### PR TITLE
add bounding box support

### PR DESCRIPTION
**Pull request by wavemechanics**
### What

Add bbox= query parameter for API and lambda, and --bbox cli parameter.
bbox can be used instead of rows= to select all LSOAs which intersect with the bounding box.
bbox argument is of the form lat1,lon1,lat2,lon2

### CODE REVIEW

File: pkg/demo/demo.go [hunk 1]
```diff
@@ -100,6 +87,75 @@ AND (
...
+		return "", fmt.Errorf("scanning bbox %q: %w", bbox, err)
+	}
+	if fields != 4 {
+		return "", ErrMissingParams
+	}
+
+	// Construct SQL
+	//
+	template := `
+SELECT
```
Comment (VivianAllen) [inline pkg/demo/demo.go:108]: is there any handy string function that could remove common indents from this multiline string and avoid breaking the indendation here? something equivalnent to [python's textwrap.dedent](https://docs.python.org/3/library/textwrap.html#textwrap.dedent)?

File: pkg/demo/demo.go [hunk 2]
```diff
@@ -174,6 +230,24 @@ AND (
 	return body.String(), nil
 }
 
+// condition wraps the output of WherePart inside "AND (...)".
+// We "know" this condition will not be the first condition in the query.
+func condition(col string, args []string) (string, error) {
```
Comment (VivianAllen) [inline pkg/demo/demo.go]: I think additionalCondition might be a clearer name?

Comment (VivianAllen) [reply pkg/demo/demo.go]: or extraCondition? (or something)

**Review by VivianAllen — CHANGES_REQUESTED** [review]
looks really nice! Only thing I'd say is the name of the `condition` function could be clearer, seeing as it only deals with adding extra conditions?

File: pkg/demo/demo.go (hunk 2 above)
Comment (wavemechanics) [reply pkg/demo/demo.go]: everybody else is better at names than I am :)
fixed

Comment (VivianAllen) [reply pkg/demo/demo.go]: 👍

**Review by VivianAllen — APPROVED** [review]
lgtm ✅

File: pkg/demo/demo.go (hunk 1 above)
Comment (stmuk) [reply pkg/demo/demo.go:108]: That's common practice in go. Also most have gofmt running on the code automatically when it writes a file.

**Pull request merged**